lib/maths/CMultinomialConjugate.cc

```diff
@@ -660,25 +659,24 @@ CMultinomialConjugate::jointLogMarginalLikelihood(const TDouble1Vec& samples,
         LOG_TRACE(<< "# samples = " << numberSamples
                   << ", total concentration = " << m_TotalConcentration);
 
-        result = boost::math::lgamma(numberSamples + 1.0) +
-                 boost::math::lgamma(m_TotalConcentration) -
-                 boost::math::lgamma(m_TotalConcentration + numberSamples);
+        result = std::lgamma(numberSamples + 1.0) + std::lgamma(m_TotalConcentration) -
+                 std::lgamma(m_TotalConcentration + numberSamples);
```

Comment:
We've lost the throw on failure here. Whilst this definitely shouldn't overflow, I think I'd still prefer to keep checks for NaN and finite values as per other distribution models and return accordingly. Also, try catch redundant here.

Reply:
this is a bit hidden: if you expand the diff you see at the end:

```
maths_t::EFloatingPointErrorStatus status = CMathsFuncs::fpStatus(result);

return status
```

Therefore there was no need to add another check.

Reply:
Are yes, missed that.